src/imex.rs

```diff
@@ -707,22 +710,29 @@ fn export_self_keys(context: &Context, dir: impl AsRef<Path>) -> Result<()> {
 /*******************************************************************************
  * Classic key export
  ******************************************************************************/
-fn export_key_to_asc_file(
+fn export_key_to_asc_file<T>(
     context: &Context,
     dir: impl AsRef<Path>,
     id: Option<i64>,
-    key: &Key,
-) -> std::io::Result<()> {
+    key: &T,
+) -> std::io::Result<()>
+where
+    T: DcKey + Any,
+{
     let file_name = {
-        let kind = if key.is_public() { "public" } else { "private" };
+        let any_key = key as &dyn Any;
+        let kind = match any_key.downcast_ref::<SignedPublicKey>() {
+            Some(_pubkey) => "public",
+            None => "private",
+        };
```

Comment:
This is probably the most exciting part of this PR.  It assume knowledge that DcKey will only ever be implemented by these two types which is ok I guess.  The alternative was to move this function to a method of DcKey, but that's ugly too since this wrapper would still be needed for the error/callback as they certainly don't belong in key.rs.

Reply:
instead of assuming that we only ever have two implementors, I would do sth like this

```rust
let kind = if any_key.downcast_ref::<SignedPublicKey>().is_some() {
  "public"
} else if any_key.downcast_ref::<SignedSecretKey>().is_some() {
  "private"
} else {
  "unknown"
}
```

Reply:
+1